internal/guest/storage/scsi/scsi.go

```diff
@@ -159,6 +159,11 @@ func ControllerLunToName(ctx context.Context, controller, lun uint8) (_ string,
 		if len(deviceNames) == 0 {
 			select {
 			case <-ctx.Done():
+				if scsiDevices, iErr := ioutil.ReadDir("/sys/bus/scsi/devices"); iErr != nil {
+					log.G(ctx).WithError(iErr).Debug("failed to list scsi devices")
+				} else {
+					log.G(ctx).WithField("scsiDevices", scsiDevices).Debug("scsi devices at context timeout")
```

Comment:
Logging just the `scsiDevices` directly would just print the addresses of the slice elements. For example I tried this locally with a directory and got:

entries="[0xc0000e4070 0xc0000e40e0 0xc0000e4150 0xc0000e41c0 0xc0000e4230 0xc0000e42a0 0xc0000e4310 0xc0000e4380 0xc0000e43f0 0xc0000e4460 0xc0000e44d0 0xc0000e4540 0xc0000e45b0 0xc0000e4620 0xc0000e4690 0xc0000e4700 0xc0000e4770 0xc0000e47e0 0xc0000e4850 0xc0000e48c0 0xc0000e4930 0xc0000e49a0 0xc0000e4a10 0xc0000e4a80 0xc0000e4af0 0xc0000e4b60 0xc0000e4bd0 0xc0000e4c40 0xc0000e4cb0 0xc0000e4d20 0xc0000e4d90 0xc0000e4e00 0xc0000e4e70]

Reply:
thanks for checking this. I naively assumed the structs will be printed. I'll update.

Reply:
I always forget what types the different ioutil functions return so I've done this quite a lot.. 😆